src/crispy/CMakeLists.txt

```diff
@@ -41,7 +41,7 @@ else()
     list(APPEND CRISPY_CORE_LIBS ${FILESYSTEM_LIBS})
 endif()
 
-target_compile_options(crispy-core PUBLIC -maes)
+target_compile_options(crispy-core PUBLIC -march=armv8-a+fp+simd+crypto+crc)
```

Comment:
The initial solution (just adding `-maes`) was wrong to begin with. On Windows/MSVC for example this always creates compiler warnings. I am not sure about this change yet (while needed), maybe put it between `if()` conditions? we can also conditionally enable iff the target compiler supports that.

Reply:
EDIT: Now I've been reading your question. I may come up with a suggestion ASAP (currently busy), maybe someone else's faster otherwise? :)

Reply:
Looks like https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html can be used, such as:

```cmake
if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
    target_compile_options(crispy-core PUBLIC -maes)
elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64) # ARM64
    target_compile_options(crispy-core PUBLIC -march=armv8-a+fp+simd+crypto+crc)
else()
    message(FATAL_ERROR "Target architecture not detected.")
endif()
```

Actual usecase I found on the net: https://github.com/Samsung/netcoredbg/blob/master/detectplatform.cmake#L22-L38